### ISSUE TITLE
MediaType Validation does not support case insensitive Charset

### ISSUE THREAD

**Issue by tobiasvoelke** [cdimascio/express-openapi-validator#463]
**Describe the bug**
According to [RFC2046](https://tools.ietf.org/html/rfc2046#section-4.1.2) the charset in media types should be case insensitive. Whenever I pass a `content-type` where the charset does not match the case of the api spec I receive `unsupported media type <...>`. In case of a uppercase charset in the api spec even setting only the media type will fail

**To Reproduce**

1. Provide API Spec containing a `content-type` with charset like e.g.  'application/json; charset=UTF-8'
    Send a request with `content-type`  'application/json; charset=utf-8' or  'application/json'

2. Provide API Spec containing a `content-type` with charset like e.g.  'application/json; charset=utf-8'
    Send a request with `content-type`  'application/json; charset=UTF-8'

**Actual behavior** 
Media Type validation fails with `unsupported media type <...>`

**Expected behavior**
Should accept charsets that only differ in casing

**Examples and context**
```
const express = require('express');
const bodyParser = require('body-parser')

const app = express();
const OpenApiValidator = require('express-openapi-validator');

app.use(bodyParser.json());

app.use(
  OpenApiValidator.middleware({
    apiSpec: 'api.yaml',
    validateResponses: true,
  }),
);

app.post('/api/foo', function (req, res, next) {
  res.json({});
});

app.listen(8080)
```

```
openapi: '3.0.3'
info:
  title: Dummy
  version: '0.1.0'

paths:
  /api/foo:
    post:
      requestBody:
        content:
          'application/json; charset=utf-8':
            schema:
              type: object
      responses:
        200:
          description: OK
          content:
            'application/json; charset=utf-8':
              schema:
                type: object
```

```
POST http://localhost:8080/api/foo
Content-Type: application/json; charset=UTF-8

{"foo":  "test"}
```

Providing the `content-type` as  'application/json; charset=UTF-8' in the spec will also cause API calls with `content-type`: `application/json` to fail.

```
openapi: '3.0.3'
info:
  title: Dummy
  version: '0.1.0'

paths:
  /api/foo:
    post:
      requestBody:
        content:
          'application/json; charset=UTF-8':
            schema:
              type: object
      responses:
        200:
          description: OK
          content:
            'application/json; charset=UTF-8':
              schema:
                type: object
```

```
POST http://localhost:8080/api/foo
Content-Type: application/json

{"foo":  "test"}
```

**Comment by cdimascio** [comment]
@tobiasvoelke thanks for the issue. definitely something to fix.

**Comment by cdimascio** [comment]
fixed in `v4.10.1`

**Comment by jayakrishnar** [comment]
The issue back again in version 4.13.7
